software/bsg_manycore_lib/bsg_striped_array.hpp

```diff
@@ -0,0 +1,69 @@
+#pragma once
+extern "C" {
+#include "bsg_manycore.h"
+}
+
+namespace bsg_manycore {
+    template <typename TYPE, int SIZE, int TG_SIZE_X, int TG_SIZE_Y, int STRIPE_SIZE=1>
+    class TileGroupStripedArray {
+    public:
+        static constexpr int TILES = TG_SIZE_X * TG_SIZE_Y;
+        static constexpr int STRIPES = SIZE/STRIPE_SIZE + SIZE%STRIPE_SIZE;
+        static constexpr int STRIPES_PER_TILE = (STRIPES/TILES) + (STRIPES%TILES == 0 ? 0 : 1);
+        static constexpr int ELEMENTS_PER_TILE = STRIPES_PER_TILE * STRIPE_SIZE;
+
+        TileGroupStripedArray() {}
+
+        int size() const { return SIZE; }
+        int stripe_size() const { return STRIPE_SIZE; }
+
+        TYPE operator[](int i) const {
+            return *_address(i);
+
+        }
+
+        TYPE & operator[](int i) {
+            return *_address(i);
+        }
+
+        TYPE at_local(int i) const {
```

Comment:
What is `at_local` for? Seems like it takes away from the abstraction of having this be a simple array.

Reply:
It's to access the ith entry on the current tile. It seemed like something I would want so I put it there.

Reply:
Fair enough